### PR TITLE
[POAE7-1474] Implement data conversion round trip between velox and omnisci with arrow for basic types

### PR DESCRIPTION
**Pull request by yyang52**
convert velox vector to cider col buffer through arrow. As velox to arrow is already implemented as exportToArrow, directly use that function and implement arrow to cider. Here zero copy seems hard as arrow array's buffer is set to const but we need change the null represent for cider side.

### CODE REVIEW

File: velox/omnisci/ArrowDataConvertor.cpp [hunk 1]
```diff
@@ -13,22 +13,303 @@
  */
 #include "velox/omnisci/ArrowDataConvertor.h"
 namespace facebook::velox::cider {
+template <TypeKind kind>
+void toCiderWithArrowImpl(
```
Comment (winningsix) [inline velox/omnisci/ArrowDataConvertor.cpp:17]: In this API, we don't care about Velox format. It's not something converting from Cider to Omnisci format with Velox as an intermediate format. This set of API is to accepting format already converted to Arrow. SO it's input parameter is just ArrowArray without Velox VectorPtr.

**Review by winningsix — COMMENTED** [review]
A few comments left. Major comment is we need a set of public APIs accepting Arrow format.

File: velox/omnisci/ArrowDataConvertor.cpp (hunk 1 above)
Comment (yyang52) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: So here actually what we need to implement conversion between Arrow and omnisci without considering velox?

Comment (winningsix) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: I think so. B/c we already have VectorBridge implemented by Velox path. And more important, this set of API is to support other engines like Arrow. Your thoughts?

Comment (yyang52) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: That's right. Would such API still in velox side? or velox can call such api to first convert omnisci to arrow, then use its bridge function to convert to velox vector?

Comment (winningsix) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: Not in ```Velox``` but in either ```VeloxPlugin``` or ```Omnisci```. We have project spin-off internally. So you can keep it as now in this PR. Later, we will do code refactor.

Comment (yyang52) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: Sure

Comment (winningsix) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: To be more clear, please only cover Arrow to Omnisci data conversion in this PR. And we will merge it soon.

Comment (yyang52) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: means not need round trip in this PR right?

Comment (winningsix) [reply velox/omnisci/ArrowDataConvertor.cpp:17]: Oh. Sorry, still round trip but not from Velox to Arrow to Omnisci.

File: velox/omnisci/tests/ResultConvertTest.cpp [hunk 2]
```diff
@@ -192,13 +192,69 @@ TEST_F(ResultConvertTest, directToCiderTimestampOneCol) {
...
+      } else {
+        VELOX_NYI("Conversion is not supported yet");
+      }
+    } else {
+      EXPECT_EQ(data[idx], col_0[idx]);
+    }
+  }
+}
+
+TEST_F(ResultConvertTest, toCiderIntegerOneColArrow) {
```
[the remark this answers is not in the document]
Comment (yyang52) [reply velox/omnisci/tests/ResultConvertTest.cpp:224]: I could not quite get why we need TEST_P here. As TEST_P does not make the code simpler here I guess, if that's the concern?

[the remark this answers is not in the document]
Comment (winningsix) [reply velox/omnisci/tests/ResultConvertTest.cpp:224]: It's for parmaterized test.

[the remark this answers is not in the document]
Comment (yyang52) [reply velox/omnisci/tests/ResultConvertTest.cpp:224]: Yes, I check the doc and code example, just not sure why that needed?

**Pull request closed without merging**